.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,340 @@
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+  checkstyle:
+    name: Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Checkstyle
+        run: |
+          ./gradlew checkStyleMain
+          ./gradlew checkstyleTest
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Checkstyle Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Delete Temporary Artifacts
+        run: rm -r ./artifacts
+
+  spotbugs:
+    name: Spotbugs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Spotbugs
+        run: |
+          ./gradlew spotbugsMain
+          ./gradlew spotbugsTest
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Spotbugs Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Delete Temporary Artifacts
+        run: rm -r ./artifacts
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    needs: [checkstyle, spotbugs]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Create Temporary Artifacts Directory
+        run: mkdir artifacts
+
+      - name: Execute Unit Tests
+        run: ./gradlew test
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        run: cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Unit Test Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Delete Temporary Artifacts
```

Comment:
I think "Temporary Artifacts Cleanup" sounds slightly better.
Also, would suggest running "rm -r ./artifacts" with "f" as it is more error-tolerant.
`run: rm -rf ./artifacts`

Reply:
I agree!